cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -122,6 +123,18 @@ func WriteKubeletConfigFiles(cfg *kubeadmapi.InitConfiguration, patchesDir strin
 	}
 
 	errs := []error{}
+
+	if features.Enabled(cfg.FeatureGates, features.NodeLocalCRISocket) {
+		containerRuntimeEndpoint, err := kubeletphase.ReadKubeadmFlags(filepath.Join(kubeletDir, kubeadmconstants.KubeletEnvFileName), out)
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "error reading kubeadm flags file"))
+		}
+
+		if err := kubeletphase.WriteInstanceConfigToDisk(containerRuntimeEndpoint, kubeletDir); err != nil {
+			errs = append(errs, errors.Wrap(err, "error writing instance kubelet configuration to disk"))
+		}
+	}
```

Comment:
Should we handle the dryrun mode here? 
- print a dryrun log for the expected behavior.